### DISCUSSION TITLE
nroots

### DISCUSSION THREAD

**Discussion by MSAfify** [danielver02/ALPS#163] [Q&A] [answered]
Hi Prof. Daniel,
Please let me know what I should change to be suitable for nroots=2, i.e., clarify this statement, "Need to have number of name lists equal to nroots."

* Also, when I consider nroots=2 and I get such roots, how can I distinguish between their types, like in your article ALPS: the Arbitrary Linear Plasma Solver (figure 1a)?

Thanks!
Mahmoud

**Comment by danielver02** [comment, marked as answer]
Hi Mahmoud,

If you want to use nroots=2, then you need to add a block like this:
```
&guess_2
g_om=1.5508d-01   !real frequency
g_gam=-1.3429d-05 !imaginary frequency
/
```
to initialise your second root. The initial guess in this block then defines what second mode you scan.

Distinguishing roots is often a question of experience. Ultimately, all roots are equally valid solutions of the dispersion relation. This means that you can only associate them with different characteristics based on, say, analytical expectations or background knowledge of the waves. For instance, a parallel propagating Alfvén wave usually shows the cutoff at the proton cyclotron frequency. So, if you have a parallel propagating solution that starts at omega~kpar*vA and then turns towards lower phase speeds as k increases, then this is most likely the Alfvén wave branch.

You can also look at the eigenfunctions, which ALPS gives you as part of its solutions. There you can see the differences between, say, right-hand and left-hand circularly polarised modes. This also helps to distinguish them. Likewise, you can look at the plasma behaviour in these modes (are the compressive or not, which components of the velocity have the largest amplitude, etc.?). 

Best wishes,
-Daniel

**Reply by MSAfify** [reply]
Hi Prof. Daniel,
Thanks so much!
Mahmoud
